apis/Google.Cloud.Diagnostics.AspNetCore3/docs/index.md

```diff
@@ -0,0 +1,204 @@
+{{title}}
+
+`Google.Cloud.Diagnostics.AspNetCore` is an ASP.NET Core instrumentation library for Google Stackdriver.
+It allows for simple integration of Stackdriver into ASP.NET applications with minimal code changes.
+
+This package supports Stackdriver Error Reporting, Stackdriver
+Logging and Stackdriver Trace, and integrates with ASP.NET Core 2.1
+(which is the long-term support release of ASP.NET Core 2.x) and above.
+
+{{version}}
+
+{{installation}}
+
+{{auth}}
+
+See [API Permissions for `entries.write`](https://cloud.google.com/logging/docs/access-control#api-permissions)
+for the permissions needed for Logging and Error Reporting.
+
+See [API Permissions for PatchTraces](https://cloud.google.com/trace/docs/iam#api_permissions)
+for the permissions needed for Tracing.
+
+# Note
+The Google.Cloud.Diagnostics.AspNetCore package attempts to collect the filename and line number when
+entries are collected. However, to be able to collect this information PDBs must be included with
+the deployed code.
+
+# Note
+When running on environments that limit or disable CPU usage for background activities, for instance
+[Google Cloud Run](https://cloud.google.com/run/docs/tips/general#avoiding_background_activities), take care
+not to use the timed buffer options for any of Logging, Tracing or Error Reporting. Take into account
+that the timed buffer is used for all of these components by default so you will need to explicitly
+configure the buffers by using the `Google.Cloud.Diagnostics.AspNetCore.LoggerOptions`,
+`Google.Cloud.Diagnostics.Common.TraceOptions` and `Google.Cloud.Diagnostics.Common.ErrorReportingOptions` classes.
+Below you'll find examples of how to configure the buffers.
+
+# Getting started
+
+## Initializing Google Diagnostics
+
+The easiest way to initialize Google Diagnostics services is using
+the `UseGoogleDiagnostics` extentension method on `IWebHostBuilder`.
+This configures Logging, Tracing and Error Reporting middleware.
+
+If your application is runnng on Google App Engine, Google
+Kubernetes Engine, Google Cloud Run or Google Compute Engine, you
+don't need to provide a value for `ProjectID`, `Service` and
```

Comment:
Is it ProjectID or ProjectId? (I'd expect the latter.) If this needs correcting, search-and-replace it - I haven't looked for all occurrences.